pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -302,6 +302,16 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 				r.status.SetDegraded("Unable to get or create the tunnel secret", err.Error())
 				return reconcile.Result{}, err
 			}
+
+			// verify tls configurations are valid
+			if managementCluster.Spec.TLS.SecretName != "" &&
```

Comment:
I think this can go away if we add the CRD validation.

We usually write back defaults to the CRD if values are not provided. If you check for "filldefaults" in this repo, you will find some examples.